### PR TITLE
ci(schema): scheduled updates

### PR DESCRIPTION
**Pull request by hoffa**
### Description of changes

Proposes PRs on weekdays if `make schema-all` changes anything.

Ensures we stay on top of upstream changes to the CloudFormation schema.

### Description of how you validated changes

- Example PR: https://github.com/hoffa/serverless-application-model/pull/7
- With change: https://github.com/hoffa/serverless-application-model/actions/runs/4069451110
- Without change: https://github.com/hoffa/serverless-application-model/actions/runs/4069463225

### Checklist

- [x] Adheres to the [development guidelines](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#development-guidelines)
- [ ] Add/update [transform tests](https://github.com/aws/serverless-application-model/blob/develop/DEVELOPMENT_GUIDE.md#unit-testing-with-multiple-python-versions)
    - [ ] Using correct values
    - [ ] Using wrong values
- [ ] Add/update [integration tests](https://github.com/aws/serverless-application-model/blob/develop/INTEGRATION_TESTS.md)

### Examples?

Please reach out in the comments if you want to add an example. Examples will be 
added to `sam init` through [aws/aws-sam-cli-app-templates](https://github.com/aws/aws-sam-cli-app-templates).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by acristin — APPROVED** [review]
Very neat

File: .github/workflows/schema.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      - uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+
+      - name: Update schema
+        id: schema
+        run: |
+          make init
+          make schema-all
+          git diff --exit-code || echo "changed=true" >> $GITHUB_OUTPUT
```
Comment (aahung) [inline .github/workflows/schema.yml:28]: what does this line do? if only there is more comment here

File: .github/workflows/schema.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  schedule:
+    - cron: "0 0 * * 1-5" # Weekdays
+
+jobs:
+  update:
+    if: github.repository == "aws/serverless-application-model"
+    runs-on: ubuntu-latest
+
+    permissions:
+      contents: write
```
Comment (aahung) [inline .github/workflows/schema.yml:14]: What does this `contents` permission do?

File: .github/workflows/schema.yml (hunk 1 above)
Comment (hoffa) [reply .github/workflows/schema.yml:28]: Checks whether anything changed. I thought the `diff` and `changed` would be clear enough

Comment (aahung) [reply .github/workflows/schema.yml:28]: I think I am getting it after reading the `if` below

File: .github/workflows/schema.yml (hunk 2 above)
Comment (hoffa) [reply .github/workflows/schema.yml:14]: Allows to push commits: https://docs.github.com/en/actions/security-guides/automatic-token-authentication#permissions-for-the-github_token

File: .github/workflows/schema.yml (hunk 1 above)
Comment (aahung) [reply .github/workflows/schema.yml:28]:
```suggestion
          # Set gh action condition "steps.schema.outputs.changed" to true if there is diff:
          git diff --exit-code || echo "changed=true" >> $GITHUB_OUTPUT
```

File: .github/workflows/schema.yml (hunk 2 above)
Comment (aahung) [reply .github/workflows/schema.yml:14]: Got it 👍

**Review by GavinZZ — APPROVED** [review]

**Review by aahung — APPROVED** [review]

**Review by awood45 — APPROVED** [review]

**Pull request merged**